### PR TITLE
Added AsyncLocalStorage;

### PR DESCRIPTION
**Pull request by iushakov-om**
Added methodWrapAls;

### CODE REVIEW

File: src/utils/misc.ts [hunk 1]
```diff
@@ -51,3 +53,19 @@ export function errorFromHeaders(
 
   return undefined;
 }
+
+export function getSendHeaders(options?: SendOptions): Headers {
```
Comment (Aggtaa) [inline src/utils/misc.ts:57]: this method does not need the whole SendOptions object, only the headers field

File: src/utils/misc.ts [hunk 2]
```diff
@@ -51,3 +53,19 @@ export function errorFromHeaders(
...
+  const allHeaders = [];
+
+  if (options?.headers)
+    for (const [k, v] of options.headers)
+      if (k !== THREAD_CONTEXT_KEY_ADDITIONAL_HEADERS)
+        allHeaders.push([k, v]);
```
Comment (Aggtaa) [inline src/utils/misc.ts:63]: use Array.filter instead of this cycle

File: src/utils/misc.ts (hunk 1 above)
Comment (Aggtaa) [inline src/utils/misc.ts:57]: rename getSendHeaders to addThreadContextHeaders

File: src/utils/misc.ts [hunk 3]
```diff
@@ -51,3 +53,19 @@ export function errorFromHeaders(
...
+
+  const store = threadContext.getStore();
+
+  if (store)
+    allHeaders.push(...(store.get(THREAD_CONTEXT_KEY_ADDITIONAL_HEADERS) ?? []));
```
Comment (Aggtaa) [inline src/utils/misc.ts:68]: store?.get can replace the if statement

File: src/utils/misc.ts (hunk 1 above)
Comment (iushakov-om) [reply src/utils/misc.ts]: Fixed

File: src/utils/misc.ts (hunk 2 above)
Comment (iushakov-om) [reply src/utils/misc.ts]: Refactored to Array.filter()

File: src/utils/misc.ts (hunk 1 above)
Comment (iushakov-om) [reply src/utils/misc.ts]: Renamed

File: src/utils/misc.ts (hunk 3 above)
Comment (iushakov-om) [reply src/utils/misc.ts]: Fixed

**Review by Aggtaa — APPROVED** [review]

**Pull request merged**